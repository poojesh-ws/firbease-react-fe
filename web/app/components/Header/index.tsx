/**
 *
 * Header
 *
 */
import React from 'react';
import { Layout } from 'antd';
import styled from 'styled-components';
import { fonts, colors, media } from '@themes/index';
import T from '@components/T';
import logo from '@images/icon-512x512.png';
import { Link } from 'react-router-dom';

const StyledHeader = styled(Layout.Header)`
  && {
    &.ant-layout-header {
      padding: 0 1rem;
      height: ${(props) => props.theme.headerHeight};
      align-items: center;
      justify-content: center;
      background-color: ${colors.primary};
      gap: 1rem;
      ${media.lessThan('mobile')`
      padding-left: ${(props) => props.theme.sidebarWidth}
      `}
    }
    display: flex;
  }
`;

const Logo = styled.img`
  height: 5rem;
  width: auto;
  object-fit: contain;
  ${media.lessThan('tablet')`
    height: 4rem;
  `}
`;

const Title = styled(T)`
  && {
    margin-bottom: 0;
    ${fonts.dynamicFontSize(fonts.size.xRegular, 1, 0.5)};
    display: flex;
    align-self: center;
    color: ${colors.secondaryText};
  }
`;

const Header: React.FC = () => {
  return (
    <StyledHeader data-testid="header">
      <Link to="/">
        <Logo alt="logo" src={logo} />
      </Link>
      <Title type="heading" id="wednesday_solutions" />
    </StyledHeader>
  );
};

export default Header;
