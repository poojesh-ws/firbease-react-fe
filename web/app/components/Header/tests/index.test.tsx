/**
 *
 * Tests for Header
 *
 */

import React from 'react';
import { renderProvider } from '@utils/testUtils';
import Header from '../index';

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Header />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain logo', () => {
    const { getAllByAltText } = renderProvider(<Header />);
    expect(getAllByAltText('logo').length).toBe(1);
  });
});
