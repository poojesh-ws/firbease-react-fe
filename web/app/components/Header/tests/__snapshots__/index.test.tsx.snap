// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<body>
  <div>
    <header
      class="ant-layout-header Header__StyledHeader-sc-1oujkb8-0 gUdnlM"
      data-testid="header"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="Header__Logo-sc-1oujkb8-1 jCVyTI"
          src="IMAGE_MOCK"
        />
      </a>
      <p
        class="T__StyledText-znbtqz-0 gcpeYe Header__Title-sc-1oujkb8-2 fbBQMG"
        data-testid="t"
      >
        Wednesday Solutions
      </p>
    </header>
  </div>
</body>
`;
